### ISSUE TITLE
Puppet installation fails on Debian 8

### ISSUE THREAD

**Issue by abrose** [petems/puppet-install-shell#41]
Installation fails on a Debian 8 base box [https://github.com/holms/vagrant-jessie-box/releases/download/Jessie-v0.1/Debian-jessie-amd64-netboot.box].
The host runs Win 8.1 with the latest vagrant version (1.7.2)

```
$ vagrant provision
==> default: Installing Puppet 4.0.0 package...
==> default: 08:23:01 +0000 INFO: Version parameter defined: 4.0.0
==> default: 08:23:01 +0000 INFO: Downloading Puppet 4.0.0 for debian...
==> default: 08:23:01 +0000 INFO: Debian platform! Lets get you a DEB...
==> default: 08:23:01 +0000 INFO: Downloading http://apt.puppetlabs.com/puppetlabs-release-jessie.deb
==> default: 08:23:01 +0000 INFO:   to file /tmp/install.sh.1774.56674/puppetlabs-release-jessie.deb
==> default: 08:23:01 +0000 INFO: Trying wget...
==> default: 08:23:02 +0000 INFO: installing with dpkg...
==> default: (Reading database ... 47826 files and directories currently installed.)
==> default: Preparing to unpack .../puppetlabs-release-jessie.deb ...
==> default: Unpacking puppetlabs-release (1.0-11) over (1.0-11) ...
==> default: Setting up puppetlabs-release (1.0-11) ...
==> default: Hit http://security.debian.org jessie/updates InRelease
==> default: Hit http://ftp.uk.debian.org jessie InRelease
==> default: Hit http://ftp.uk.debian.org jessie-updates InRelease
==> default: Hit http://security.debian.org jessie/updates/main Sources
==> default: Hit http://ftp.uk.debian.org jessie/main Sources
==> default: Hit http://security.debian.org jessie/updates/contrib Sources
==> default: Hit http://ftp.uk.debian.org jessie/contrib Sources
==> default: Hit http://security.debian.org jessie/updates/non-free Sources
==> default: Hit http://ftp.uk.debian.org jessie/non-free Sources
==> default: Hit http://security.debian.org jessie/updates/main amd64 Packages
==> default: Hit http://ftp.uk.debian.org jessie/main amd64 Packages
==> default: Hit http://security.debian.org jessie/updates/contrib amd64 Packages
==> default: Hit http://ftp.uk.debian.org jessie/contrib amd64 Packages
==> default: Hit http://security.debian.org jessie/updates/non-free amd64 Packages
==> default: Hit http://security.debian.org jessie/updates/contrib Translation-en
==> default: Hit http://ftp.uk.debian.org jessie/non-free amd64 Packages
==> default: Hit http://security.debian.org jessie/updates/main Translation-en
==> default: Hit http://ftp.uk.debian.org jessie/contrib Translation-en
==> default: Hit http://security.debian.org jessie/updates/non-free Translation-en
==> default: Hit http://ftp.uk.debian.org jessie/main Translation-en
==> default: Hit http://ftp.uk.debian.org jessie/non-free Translation-en
==> default: Ign http://apt.puppetlabs.com jessie InRelease
==> default: Hit http://ftp.uk.debian.org jessie-updates/main Sources
==> default: Hit http://ftp.uk.debian.org jessie-updates/contrib Sources
==> default: Hit http://ftp.uk.debian.org jessie-updates/non-free Sources
==> default: Hit http://apt.puppetlabs.com jessie Release.gpg
==> default: Get:1 http://ftp.uk.debian.org jessie-updates/main amd64 Packages/DiffIndex [367 B]
==> default: Hit http://ftp.uk.debian.org jessie-updates/contrib amd64 Packages
==> default: Hit http://ftp.uk.debian.org jessie-updates/non-free amd64 Packages
==> default: Hit http://apt.puppetlabs.com jessie Release
==> default: Hit http://ftp.uk.debian.org jessie-updates/contrib Translation-en
==> default: Get:2 http://ftp.uk.debian.org jessie-updates/main Translation-en/DiffIndex [229 B]
==> default: Hit http://ftp.uk.debian.org jessie-updates/non-free Translation-en
==> default: Hit http://apt.puppetlabs.com jessie/main Sources
==> default: Hit http://apt.puppetlabs.com jessie/dependencies Sources
==> default: Hit http://apt.puppetlabs.com jessie/main amd64 Packages
==> default: Hit http://apt.puppetlabs.com jessie/dependencies amd64 Packages
==> default: Ign http://apt.puppetlabs.com jessie/dependencies Translation-en_US
==> default: Ign http://apt.puppetlabs.com jessie/dependencies Translation-en
==> default: Ign http://apt.puppetlabs.com jessie/main Translation-en_US
==> default: Ign http://apt.puppetlabs.com jessie/main Translation-en
==> default: Fetched 596 B in 2s (199 B/s)
==> default: Reading package lists...

==> default: Reading package lists...

==> default: Building dependency tree...
==> default:
==> default: Reading state information...

==> default: E: Version '4.0.0-1puppetlabs1' for 'puppet-common' was not found
==> default: E: Version '4.0.0-1puppetlabs1' for 'puppet' was not found
==> default: 08:23:10 +0000 CRIT: Installation failed
==> default: 08:23:10 +0000 CRIT: Please file a bug report at https://github.com/petems/puppet-install-shell/
==> default: 08:23:10 +0000 CRIT:
==> default: 08:23:10 +0000 CRIT: Version: 4.0.0
==> default: 08:23:10 +0000 CRIT: Platform: debian
==> default: 08:23:10 +0000 CRIT: Platform Version: 8.0
==> default: 08:23:10 +0000 CRIT: Machine: x86_64
==> default: 08:23:10 +0000 CRIT: OS: Linux
==> default: 08:23:10 +0000 CRIT:
==> default: 08:23:10 +0000 CRIT: Please detail your operating system type, version and any other relevant details
The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

sh install.sh -v 4.0.0 2>&1

Stdout from the command:

08:23:01 +0000 INFO: Version parameter defined: 4.0.0
08:23:01 +0000 INFO: Downloading Puppet 4.0.0 for debian...
08:23:01 +0000 INFO: Debian platform! Lets get you a DEB...
08:23:01 +0000 INFO: Downloading http://apt.puppetlabs.com/puppetlabs-release-jessie.deb
08:23:01 +0000 INFO:   to file /tmp/install.sh.1774.56674/puppetlabs-release-jessie.deb
08:23:01 +0000 INFO: Trying wget...
08:23:02 +0000 INFO: installing with dpkg...
(Reading database ... 47826 files and directories currently installed.)
Preparing to unpack .../puppetlabs-release-jessie.deb ...
Unpacking puppetlabs-release (1.0-11) over (1.0-11) ...
Setting up puppetlabs-release (1.0-11) ...
Hit http://security.debian.org jessie/updates InRelease
Hit http://ftp.uk.debian.org jessie InRelease
Hit http://ftp.uk.debian.org jessie-updates InRelease
Hit http://security.debian.org jessie/updates/main Sources
Hit http://ftp.uk.debian.org jessie/main Sources
Hit http://security.debian.org jessie/updates/contrib Sources
Hit http://ftp.uk.debian.org jessie/contrib Sources
Hit http://security.debian.org jessie/updates/non-free Sources
Hit http://ftp.uk.debian.org jessie/non-free Sources
Hit http://security.debian.org jessie/updates/main amd64 Packages
Hit http://ftp.uk.debian.org jessie/main amd64 Packages
Hit http://security.debian.org jessie/updates/contrib amd64 Packages
Hit http://ftp.uk.debian.org jessie/contrib amd64 Packages
Hit http://security.debian.org jessie/updates/non-free amd64 Packages
Hit http://security.debian.org jessie/updates/contrib Translation-en
Hit http://ftp.uk.debian.org jessie/non-free amd64 Packages
Hit http://security.debian.org jessie/updates/main Translation-en
Hit http://ftp.uk.debian.org jessie/contrib Translation-en
Hit http://security.debian.org jessie/updates/non-free Translation-en
Hit http://ftp.uk.debian.org jessie/main Translation-en
Hit http://ftp.uk.debian.org jessie/non-free Translation-en
Ign http://apt.puppetlabs.com jessie InRelease
Hit http://ftp.uk.debian.org jessie-updates/main Sources
Hit http://ftp.uk.debian.org jessie-updates/contrib Sources
Hit http://ftp.uk.debian.org jessie-updates/non-free Sources
Hit http://apt.puppetlabs.com jessie Release.gpg
Get:1 http://ftp.uk.debian.org jessie-updates/main amd64 Packages/DiffIndex [367 B]
Hit http://ftp.uk.debian.org jessie-updates/contrib amd64 Packages
Hit http://ftp.uk.debian.org jessie-updates/non-free amd64 Packages
Hit http://apt.puppetlabs.com jessie Release
Hit http://ftp.uk.debian.org jessie-updates/contrib Translation-en
Get:2 http://ftp.uk.debian.org jessie-updates/main Translation-en/DiffIndex [229 B]
Hit http://ftp.uk.debian.org jessie-updates/non-free Translation-en
Hit http://apt.puppetlabs.com jessie/main Sources
Hit http://apt.puppetlabs.com jessie/dependencies Sources
Hit http://apt.puppetlabs.com jessie/main amd64 Packages
Hit http://apt.puppetlabs.com jessie/dependencies amd64 Packages
Ign http://apt.puppetlabs.com jessie/dependencies Translation-en_US
Ign http://apt.puppetlabs.com jessie/dependencies Translation-en
Ign http://apt.puppetlabs.com jessie/main Translation-en_US
Ign http://apt.puppetlabs.com jessie/main Translation-en
Fetched 596 B in 2s (199 B/s)
Reading package lists...
Reading package lists...
Building dependency tree...
Reading state information...
E: Version '4.0.0-1puppetlabs1' for 'puppet-common' was not found
E: Version '4.0.0-1puppetlabs1' for 'puppet' was not found
08:23:10 +0000 CRIT: Installation failed
08:23:10 +0000 CRIT: Please file a bug report at https://github.com/petems/puppet-install-shell/
08:23:10 +0000 CRIT:
08:23:10 +0000 CRIT: Version: 4.0.0
08:23:10 +0000 CRIT: Platform: debian
08:23:10 +0000 CRIT: Platform Version: 8.0
08:23:10 +0000 CRIT: Machine: x86_64
08:23:10 +0000 CRIT: OS: Linux
08:23:10 +0000 CRIT:
08:23:10 +0000 CRIT: Please detail your operating system type, version and any other relevant details


Stderr from the command:

stdin: is not a tty
```

**Comment by theninthnode** [comment]
Also receiving the same error (diff puppetlabs1 version number):

```
E: Version '4.1.0-1puppetlabs1' for 'puppet-common' was not found
E: Version '4.1.0-1puppetlabs1' for 'puppet' was not found
```

Also a Debian box. Also Vagrant V 1.7.2.

@abrose did you figure out the issue?

**Comment by abrose** [comment]
No, unfortunately not. I ended up installing puppet via  shell provider before the puppet provider is called.

**Comment by petems** [comment]
Yep, Puppet > 4.0 is delivered as the puppet-agent package. Need to refactor the code to take this into account.

**Comment by petems** [comment]
Related to #40

**Comment by petems** [comment]
This should be fixed by #44
